flask_sqlalchemy/__init__.py

```diff
@@ -512,6 +512,10 @@ def __init__(self, sa):
         self.sa = sa
 
     def __get__(self, obj, type):
+        if obj is not None and obj.__class__.__bases__[0].query is None:
```

Comment:
Still need an explanation / test of the `and ...` part of this if statement.

Reply:
Without the second part this test fails:
https://github.com/mitsuhiko/flask-sqlalchemy/blob/4e4a73ca68fc2db86b5f3d2bcb368c9495262866/tests/test_regressions.py#L40

That test was added 8 years ago in #33. Do you know if this line is just a typo in the test:
https://github.com/mitsuhiko/flask-sqlalchemy/pull/33/files#diff-9aa0c68205f6674736c522c04c66d6a0R347

Reply:
Looks like it should be SubBase that gets called. But the `and` part shouldn't be necessary since `obj` will always been None when accessing from the class and this could cause issues with models that put a mixin before the model base.

Reply:
baf83f45912bf1bca251cb027564c76262a56da9 changes from `base` to capitalized `Base`, but it should instead be `SubBase`?